onnxruntime/core/providers/coreml/builders/impl/split_op_builder.cc

```diff
@@ -43,55 +46,104 @@ Status SplitOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder,
   ORT_RETURN_IF_NOT(GetShape(*node.InputDefs()[0], data_shape, logger), "Failed to get input shape.");
 
   NodeAttrHelper helper(node);
-  const auto axis = helper.Get("axis", 0);
-
-  // attribute introduced since opset 18
-  uint64_t num_outputs;
-
-  std::unique_ptr<COREML_SPEC::NeuralNetworkLayer> layer = model_builder.CreateNNLayer(node);
-  auto* coreml_splitnd = layer->mutable_splitnd();
-  coreml_splitnd->set_axis(axis);
-
-  if (input_defs.size() > 1) {
-    // if "split" is explicitly provided as an input
-    const auto& split_tensor = *model_builder.GetInitializerTensors().at(input_defs[1]->Name());
-    Initializer unpacked_tensor(split_tensor);
-    auto split_span = unpacked_tensor.DataAsSpan<uint64_t>();
-    auto split_sizes = split_span.size();
-    num_outputs = narrow<uint64_t>(split_sizes);
-    for (size_t i = 0; i < split_sizes; i++) {
-      coreml_splitnd->add_splitsizes(split_span[i]);
-    }
-  } else if (node.SinceVersion() < 18) {
-    num_outputs = narrow<uint64_t>(node.OutputDefs().size());
-    coreml_splitnd->set_numsplits(num_outputs);
-  } else {
-    // note: for opset 18+ 'num_outputs' is a required attribute
-    num_outputs = narrow<uint64_t>(helper.GetInt64("num_outputs").value());
-    // note: checked in IsOpSupportedImpl that ensures the dim value at splitting axis exists
-    auto split_dim_size = data_shape[HandleNegativeAxis(axis, data_shape.size())];
-    uint64_t chunk_size = narrow<uint64_t>((split_dim_size + num_outputs - 1) / num_outputs);
-    uint64_t remainder = split_dim_size % chunk_size;
-    if (remainder) {
-      // uneven
-      auto split_sizes = InlinedVector<uint64_t>(num_outputs, chunk_size);
-      split_sizes.back() = remainder;
-      for (size_t i = 0; i < split_sizes.size(); i++) {
-        coreml_splitnd->add_splitsizes(split_sizes[i]);
+  int64_t axis = helper.Get("axis", 0);
+
+
+#if defined(COREML_ENABLE_MLPROGRAM)
+  if (model_builder.CreateMLProgram()) {
+      int64_t num_outputs;
+      using namespace CoreML::Specification::MILSpec;
+      std::unique_ptr<Operation> split_op = model_builder.CreateOperation(node, "split");
+      AddOperationInput(*split_op, "axis", model_builder.AddScalarConstant(split_op->type(), "axis", axis));
+
+      if (input_defs.size() > 1) {
+        // if "split" is explicitly provided as an input
+        const auto& split_tensor = *model_builder.GetInitializerTensors().at(input_defs[1]->Name());
+        Initializer unpacked_tensor(split_tensor);
+        auto split_span = unpacked_tensor.DataAsSpan<int64_t>();
+        num_outputs = narrow<int64_t>(split_span.size());
+        AddOperationInput(*split_op, "split_sizes", model_builder.AddConstant(split_op->type(), "split_sizes", split_span));
+      } else if (node.SinceVersion() < 18) {
+        num_outputs = narrow<uint64_t>(node.OutputDefs().size());
```

Comment:
```suggestion
        num_outputs = narrow<int64_t>(node.OutputDefs().size());
```

AddConstant/AddScalarConstant supports int64_t not uint64_t as dimensions and index values in ONNX are int64_t.